### PR TITLE
Feature/disabled email inputs [VID-117]

### PR DESCRIPTION
**Pull request by rafarubim**
#### What problem is this solving?
This adds `disabledEmailInputs` prop to disable editing in email fields. This will be used by checkout to lock a known user

#### How to test it?

Go into
https://rafaprtest5--storecomponents.myvtex.com/
and click `Sign in`, then `Receive access code by email` or `Sign in with email and password`. You'll see the email input disabled.
If you add the query string `?userEmail=example@email.com`, you can lock a specific email.

#### Screenshots or example usage:

image

### CODE REVIEW

**Review by psmeireles — DISMISSED** [review]

File: docs/README.md [hunk 1]
```diff
@@ -129,6 +129,7 @@ Through the Storefront, you can change the `login`'s behavior and interface. How
 | `hasGoogleOneTap`   | `boolean` | ![https://img.shields.io/badge/-Beta-red](https://img.shields.io/badge/-Beta-red) Defines whether the [Google's One-tap sign-up and auto sign-in](https://developers.google.com/identity/one-tap/web/) solution is enabled (`true`) or not (`false`) .  |  `false`  |
 | `googleOneTapAlignment`     | `enum`    |  ![https://img.shields.io/badge/-Beta-red](https://img.shields.io/badge/-Beta-red) Defines pop-up alignment for the Google One-tap login. Possible values are `Left` and `Right`. | `Right`    |
 | `googleOneTapMarginTop`  | `string`  | ![https://img.shields.io/badge/-Beta-red](https://img.shields.io/badge/-Beta-red) Defines the pop-up top margin for the Google One-tap login. The values supported are the same supported by the CSS property `top`. | `3rem`       |
+| `disabledEmailInputs`                 | `Boolean` | Disables all email inputs                                                                     | -             |
```
[the remark this answers is not in the document]
Comment (rafarubim) [reply docs/README.md]: I just included it!

File: docs/README.md [hunk 2]
```diff
@@ -129,6 +129,7 @@ Through the Storefront, you can change the `login`'s behavior and interface. How
 | `hasGoogleOneTap`   | `boolean` | ![https://img.shields.io/badge/-Beta-red](https://img.shields.io/badge/-Beta-red) Defines whether the [Google's One-tap sign-up and auto sign-in](https://developers.google.com/identity/one-tap/web/) solution is enabled (`true`) or not (`false`) .  |  `false`  |
 | `googleOneTapAlignment`     | `enum`    |  ![https://img.shields.io/badge/-Beta-red](https://img.shields.io/badge/-Beta-red) Defines pop-up alignment for the Google One-tap login. Possible values are `Left` and `Right`. | `Right`    |
 | `googleOneTapMarginTop`  | `string`  | ![https://img.shields.io/badge/-Beta-red](https://img.shields.io/badge/-Beta-red) Defines the pop-up top margin for the Google One-tap login. The values supported are the same supported by the CSS property `top`. | `3rem`       |
+| `disabledEmailInputs`                 | `Boolean` | Disables all email inputs                                                                     | false         |
```
Comment (guerreirobeatriz) [inline docs/README.md:132]:
```suggestion
| `disabledEmailInputs`     | `boolean` | Whether Checkout should allow user email editing (`true`) or not (`false`).   | `false`       |
```

File: docs/README.md [hunk 3]
```diff
@@ -150,6 +151,7 @@ You can also change the `login-content`'s behaviour and interface through the St
 | `identifierPlaceholder`               | `String`  | Set placeholder for the identifier extension                                                  | -             |
 | `invalidIdentifierError`              | `String`  | Set error message for invalid user identifier                                                 | -             |
 | `termsAndConditions`                  | `String`  | Sets a markdown text below the login options, meant to warn the user about terms & conditions | -             |
+| `disabledEmailInputs`                 | `Boolean` | Disables all email inputs                                                                     | false         |
```
Comment (guerreirobeatriz) [inline docs/README.md:154]:
```suggestion
| `disabledEmailInputs`                 | `boolean` | Whether Checkout should allow user email editing (`true`) or not (`false`).      | `false`         |
```

File: docs/README.md (hunk 2 above)
Comment (rafarubim) [reply docs/README.md:132]: Ops, forgot about that, sorry!

**Review by guerreirobeatriz — DISMISSED** [review]

File: react/components/EmailAndPassword.js [hunk 4]
```diff
@@ -46,6 +46,8 @@ class EmailAndPassword extends Component {
     identifierPlaceholder: PropTypes.string,
     /** Error message for the user identifier */
     invalidIdentifierError: PropTypes.string,
+    /** Disables editing email input */
```
Comment (rnbez) [inline react/components/EmailAndPassword.js:49]:
```suggestion
    /** Disables email input */
```

File: react/components/EmailVerification.js [hunk 5]
```diff
@@ -35,6 +35,8 @@ class EmailVerification extends Component {
     intl: intlShape,
     /** Whether to display the back button */
     showBackButton: PropTypes.bool,
+    /** Disables editing email input */
```
Comment (rnbez) [inline react/components/EmailVerification.js:38]:
```suggestion
    /** Disables email input */
```

File: react/propTypes.js [hunk 6]
```diff
@@ -44,6 +44,8 @@ export const LoginContainerProptypes = {
   accountOptionLinks: PropTypes.array,
   /** Terms and conditions text in markdown */
   termsAndConditions: PropTypes.string,
+  /** Disables editing in all email inputs */
```
Comment (rnbez) [inline react/propTypes.js:51]:
```suggestion
  /** Disables email inputs */
```

File: docs/README.md [hunk 7]
```diff
@@ -129,7 +129,7 @@ Through the Storefront, you can change the `login`'s behavior and interface. How
 | `hasGoogleOneTap`   | `boolean` | ![https://img.shields.io/badge/-Beta-red](https://img.shields.io/badge/-Beta-red) Defines whether the [Google's One-tap sign-up and auto sign-in](https://developers.google.com/identity/one-tap/web/) solution is enabled (`true`) or not (`false`) .  |  `false`  |
 | `googleOneTapAlignment`     | `enum`    |  ![https://img.shields.io/badge/-Beta-red](https://img.shields.io/badge/-Beta-red) Defines pop-up alignment for the Google One-tap login. Possible values are `Left` and `Right`. | `Right`    |
 | `googleOneTapMarginTop`  | `string`  | ![https://img.shields.io/badge/-Beta-red](https://img.shields.io/badge/-Beta-red) Defines the pop-up top margin for the Google One-tap login. The values supported are the same supported by the CSS property `top`. | `3rem`       |
-| `disabledEmailInputs`                 | `Boolean` | Disables all email inputs                                                                     | false         |
+| `disabledEmailInputs`     | `boolean` | Whether Checkout should allow user email editing (`true`) or not (`false`).   | `false`       |
```
Comment (rnbez) [inline docs/README.md:132]: Wait, what if I'm building a b2b store and I have a customized way of setting up the user's email and I don't want it to change? The Checkout is the use case for now, but certainly not the only one.

File: docs/README.md [hunk 8]
```diff
@@ -151,7 +151,7 @@ You can also change the `login-content`'s behaviour and interface through the St
 | `identifierPlaceholder`               | `String`  | Set placeholder for the identifier extension                                                  | -             |
 | `invalidIdentifierError`              | `String`  | Set error message for invalid user identifier                                                 | -             |
 | `termsAndConditions`                  | `String`  | Sets a markdown text below the login options, meant to warn the user about terms & conditions | -             |
-| `disabledEmailInputs`                 | `Boolean` | Disables all email inputs                                                                     | false         |
+| `disabledEmailInputs`                 | `boolean` | Whether Checkout should allow user email editing (`true`) or not (`false`).      | `false`         |
```
Comment (rnbez) [inline docs/README.md:154]: Same.

**Review by rnbez — CHANGES_REQUESTED** [review]

**Review by rnbez — DISMISSED** [review]

**Pull request merged**